doc/Authoring_Plugins.md

```diff
@@ -0,0 +1,168 @@
+   + single entry point, with code completion to see possible calls.
+ + Upgrade easily from one theia version to another version as API is backward compliant.
+
+cons:
+ - Need to stick to pre-defined API. It's not possible to tweak something if contribution point is not provided through API. Note that current API can be extended to support more stuff ;-)
+
+
+
+### Design
+A Theia app is composed of a core providing a set of widgets, commands, handlers, etc. for a specific functionality.
+
+Theia defines a runtime API allowing plug-ins to customize the IDE and add their behaviour to various aspects of the application. 
+
+In Theia, a plug-in has access to the API through an object named `theia` which is available in all plug-ins.
+[More details on API]](https://github.com/theia-ide/theia/blob/master/packages/plugin/API.md).
+
+There are two natures of plug-ins:
+ - Backend plug-in. If you're familiar with VS Code extensions, it's very close. The plug-in's code is running in its own process on the server side. API is called and it's the API that will send some actions on user's browser/UI to register new commands, etc. All the callbacks are executed on the server side on a dedicated process.
+ - Frontend plug-in. In that case, callbacks are executed in a worker thread on the UI/browser. These plug-ins are only authorized to use "browser compliant" modules. For example opening or writing to a file is impossible as all the code of the plug-in is running on the browser side. But this approach is helpful if you really want to have some stuff on the client side to avoid some network operations.
+
+## Prerequisites
+
+Having a running Theia instance. (v0.3.12+)
+
+## Project Layout
+
+We are going to create a repository named `theia-hello-world-plugin`
+
+To ease the setup of such a repository, a [Yeoman code generator](https://www.npmjs.com/package/%40theia%2Fgenerator-theia-plugin) exists and can be installed to scaffold the project.
+
+Generator can be installed and executed with the following commands. Note that these commands can be entered from a new terminal inside your running Theia instance.
+
+```bash
+npm install -g yo @theia/generator-theia-plugin
+mkdir theia-hello-world-plugin
+cd theia-hello-world-plugin
+yo @theia/plugin
+```
+
+![Yeoman plugin output](images/plug-ins/yeoman-plugin.png "Yeoman plugin output")
+
+Pick up default values for each questions. (default name and backend plug-in)
+
+At this step, in `theia-hello-world-plugin` folder there is a plug-in that is already built with associated source code.
+
+## Implementing the Plug-in
+
+Let's have look at the generated code now. The `package.json` contains the entry-point of your plug-in.
+
+```json
+{
+      "name": "theia-hello-world-plugin",
+      "publisher": "theia",
+      "keywords": [
+        "theia-plugin"
+      ],
+      "version": "0.0.1",
+      "files": [
+        "src"
+      ],
+      "devDependencies": {
+        "@theia/plugin": "latest",
```

Comment:
version latest is confusing, maybe provide a link to the npm page to help

Reply:
ok